### PR TITLE
New Feature: Tooltip

### PR DESCRIPTION
**Pull request by ashenden**
This PR introduces a new component to CNVS -- the Tooltip component. Tooltips are used to provide added detail to an element. Documentation has been added.

### CODE REVIEW

File: docs/_includes/components/tooltips/tooltips-interactive-demo.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  <h2>
+
+    Interactive Demo
+
+  </h2>
+
+  <p>
+
+    You can include more than body text in a message. Messages can support any HTML content. All type styling will be retained. However, if you add the class <code>.message-type-styling</code> to any typographic element, the preferred message type color and styling will be applied. For links, add the class <code>.message-link</code> to any <code>&lt;a&gt;</code>.
```
Comment (leemunroe) [inline docs/_includes/components/tooltips/tooltips-interactive-demo.html:15]: This looks like a duplication of the tooltips-positions text.

File: docs/_includes/components/tooltips/tooltips-interactive-demo.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        Tooltip on left
+
+      </button>
+
+    </div>
+
+    <div class="panel-cell panel-cell-light panel-cell-code-block">
+
+<pre class="prettyprint transparent flush lang-html">
+&lt;div class="tooltip tooltip-top"&gt;
```
Comment (leemunroe) [inline docs/_includes/components/tooltips/tooltips-interactive-demo.html:56]: Not sure if this is showing the correct example. Expected it to show me markup for the examples above (i.e. <button...>)

File: docs/_includes/components/tooltips/tooltips-positions.html [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+  <h2>
+
+    Tooltip Positions
+
+  </h2>
+
+  <p>
+
+    You can include more than body text in a message. Messages can support any HTML content. All type styling will be retained. However, if you add the class <code>.message-type-styling</code> to any typographic element, the preferred message type color and styling will be applied. For links, add the class <code>.message-link</code> to any <code>&lt;a&gt;</code>.
```
Comment (leemunroe) [inline docs/_includes/components/tooltips/tooltips-positions.html:15]: Adding `message-link` class didn't seem to change anything for me. The link was always purple. Do we want to assume purple as the default color of links or white? I would lean towards white given that I think this is our preferred style for a link in this context (the purple contrast isn't great) and that in most cases developers will forget to add the additional class.

**Comment by ashenden** [comment]
@leemunroe good catch on the notes above.  I must have copied over the markup from the `.message` component and forgot to change the text.

There is no support for `.message-link` in tooltips.

Committed a fix.

**Comment by leemunroe** [comment]
Docs look good.

Some things I noticed with the tooltip:

* The link color is purple. Is this your intent? While consistent I'm not sure about the contrast. Would prefer to inherit the white text (and keep underlined). But I don't feel that strongly about it.
* Should we have a max-width for tooltips? This gets very wide with lots of content.
* When I added the link there were some spacing and alignment issues. I think it is due to `inline-flex`.

Screenshots
image
image

**Pull request merged**